app/pathogen/arbovirus/analyze/recharts/change-in-median-seroprevalence-over-time-graph.tsx

```diff
@@ -86,16 +93,34 @@ const groupDataPointsIntoTimeBuckets = <
     };
   }
 
-  const { buckets } = generateBucketsForTimeInterval({
-    earliestDate,
-    latestDate,
-    groupingBucketSize: input.groupingBucketSize
-  })
+  if(input.validBucketSizes.length === 0) {
+    return { groupedDataPoints: [] };
+  }
+
+  const { groupedDataPoints } = input.validBucketSizes.map((bucketSize) => {
```

Comment:
This part is kind of confusing but the idea is that it generates buckets with 1 year, 2 years, 3 years, 4 years, and 5 years and then figures out how many bars each of them would have and then picks the bucket size that got closest to `desiredBucketCount`, in this case 10.